packages/flutter/lib/src/material/page.dart

```diff
@@ -36,6 +39,108 @@ class _MaterialPageTransition extends AnimatedWidget {
   }
 }
 
+// Used for iOS.
+class _CupertinoPageTransition extends AnimatedWidget {
+  static final FractionalOffsetTween _kTween = new FractionalOffsetTween(
+    begin: FractionalOffset.topRight,
+    end: -FractionalOffset.topRight
+  );
+
+  _CupertinoPageTransition({
+    Key key,
+    Animation<double> animation,
+    this.child
+  }) : super(
+    key: key,
+    animation: _kTween.animate(new CurvedAnimation(
+      parent: animation,
+      curve: new _CupertinoTransitionCurve()
+    )
+  ));
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(ianh): tell the transform to be un-transformed for hit testing
+    // but not while being controlled by a gesture.
+    return new SlideTransition(
+      position: animation,
+      child: new Material(
+        elevation: 6,
+        child: child
+      )
+    );
+  }
+}
+
+class AnimationMean extends CompoundAnimation<double> {
+  AnimationMean({
+    Animation<double> left,
+    Animation<double> right,
+  }) : super(first: left, next: right);
+
+  @override
+  double get value => (first.value + next.value) / 2.0;
+}
+
+// Custom curve for iOS page transitions. The halfway point is when the page
+// is fully on-screen. 0.0 is fully off-screen to the right. 1.0 is off-screen
+// to the left.
+class _CupertinoTransitionCurve extends Curve {
+  _CupertinoTransitionCurve();
+
+  @override
+  double transform(double t) {
+    if (t > 0.5)
+      return (t - 0.5) / 3.0 + 0.5;
+    return t;
+  }
+}
+
+// This class responds to drag gestures to control the route's transition
+// animation progress. Used for iOS back gesture.
+class _CupertinoBackGestureController extends NavigationGestureController {
+  _CupertinoBackGestureController({
+    NavigatorState navigator,
+    this.controller,
+    this.onDisposed,
+  }) : super(navigator);
+
+  @override
+  void dispose() {
+    super.dispose();
+    onDisposed();
+    controller.removeStatusListener(onStatusChanged);
+    controller = null;
+  }
+
+  @override
+  void dragUpdate(double delta) {
+    controller.value -= delta;
+  }
+
+  @override
+  void dragEnd() {
+    if (controller.value <= 0.5) {
+      navigator.pop();
+    } else {
+      controller.forward();
+    }
+    // Don't end the gesture until the transition completes.
+    onStatusChanged(controller.status);
+    controller?.addStatusListener(onStatusChanged);
+  }
+
+  void onStatusChanged(AnimationStatus status) {
```

Comment:
this should be called handleStatusChanged. The convention is onFoo is the property, handleFoo is the closure.